components/script/dom/element.rs

```diff
@@ -2245,6 +2245,20 @@ impl VirtualMethods for Element {
         }
 
         let doc = document_from_node(self);
+
+        if self.is_focusable_area() {
+            let doc = document_from_node(self);
+            let children = self.upcast::<Node>().traverse_preorder();
+
+            for child in children {
+                if let Some(element) = child.downcast::<Element>() {
+                    if element.is_focusable_area() && !element.disabled_state() {
```

Comment:
We will need to account for elements that have their disabled state modified dynamically, too.